outlook/mail/pkg/commands/listMailFolders.go

```diff
@@ -30,23 +30,35 @@ func ListMailFolders(ctx context.Context) error {
 			return fmt.Errorf("failed to create GPTScript client: %w", err)
 		}
 
-		dataset, err := gptscriptClient.CreateDataset(ctx, os.Getenv("GPTSCRIPT_WORKSPACE_DIR"), "outlook_mail_folders", "Outlook mail folders")
-		// If we got back an error, we just print the folders. Otherwise, write them to the dataset.
-		if err == nil {
-			for _, folder := range result {
-				folderStr, err := printers.MailFolderToString(folder)
-				if err != nil {
-					return fmt.Errorf("failed to convert mail folder to string: %w", err)
-				}
-
-				if _, err = gptscriptClient.AddDatasetElement(ctx, os.Getenv("GPTSCRIPT_WORKSPACE_DIR"), dataset.ID, util.Deref(folder.GetId()), util.Deref(folder.GetDisplayName()), folderStr); err != nil {
-					return fmt.Errorf("failed to add element: %w", err)
-				}
+		workspaceID := os.Getenv("GPTSCRIPT_WORKSPACE_ID")
+
+		dataset, err := gptscriptClient.CreateDataset(ctx, workspaceID, "outlook_mail_folders", "Outlook mail folders")
```

Comment:
nitL is there a reason this one has `_` and some of the others don't?

Reply:
nope not really